library/std/src/sys_common/remutex.rs

```diff
@@ -107,30 +116,6 @@ impl<T> Drop for ReentrantMutex<T> {
     }
 }
 
-impl<T: fmt::Debug + 'static> fmt::Debug for ReentrantMutex<T> {
```

Comment:
I removed the Debug implementation for ReentrantMutex because it wasn't used anywhere and it doesn't enforce the pinnedness.